src/Microsoft.OData.Core/UriParser/Parsers/UriQueryExpressionParser.cs

```diff
@@ -277,21 +284,31 @@ internal AggregateToken ParseAggregate()
 
             this.lexer.NextToken();
 
-            return new AggregateToken(statements);
+            return statements;
         }
 
-        internal AggregateExpressionToken ParseAggregateExpression()
+        internal AggregateExpressionToken ParseAggregateExpression(QueryToken parent = null)
         {
             // expression
-            var expression = this.ParseExpression();
+            QueryToken expression = ParseLogicalOr(parent, new FunctionCallParser(this.lexer, this, true /* ignoreParseErrors */));

```

Comment:
I do not like the idea too. I didn't even like it when I had to add the `token` parameter.